### ISSUE TITLE
Public API to get the size of a mesh axis

### ISSUE THREAD

**Issue by gabbard** [apple/axlearn#893]
You can use `jax.lax.axis_index(axis)` to get your index along a mesh axis without needing access to the `mesh` object.  However, there does not seem to be a way to get access to the *size* of a mesh axis through the public API.  

We have been doing:
```
        tensor_parallel_size = jax.interpreters.pxla.thread_resources.env.shape["TP"]
```

but `jax.interpreters` [is not part of the public API](https://jax.readthedocs.io/en/latest/api_compatibility.html).

Should such a function be added?  Or are we doing something wrong if we think we sometimes need it?

**Comment by gabbard** [comment]
🤦 I had the wrong repo open when I filed this; this should be JAX.
